### PR TITLE
fix(vue-app): not strip trailing slash for redirect external domain

### PR DESCRIPTION
**Pull request by clarkdo**
## Types of changes

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

Add test for https://github.com/nuxt/nuxt.js/pull/7475

When redirect to external domain, not strip trailing slash.

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [ ] I have added tests to cover my changes (if not applicable, please state why)
- [ ] All new and existing tests are passing.

### CODE REVIEW

File: packages/vue-app/template/utils.js [hunk 1]
```diff
@@ -597,6 +597,10 @@ function formatUrl (url, query) {
   let result = (protocol ? protocol + '://' : '//') + parts.shift()
 
   let path = parts.join('/')
+  if (path === '') {
+    return result + (parts.length === 1 ? '/' : '')
```
Comment (acu192) [inline packages/vue-app/template/utils.js:601]: This early return is not using incorporating the `query` parameter. I think your new logic here should be included in the logic on line 610 below (`result += path ? '/' + path : ''`).

Comment (clarkdo) [reply packages/vue-app/template/utils.js:601]: Here is only for url only has a domain without path, querystring and hash

Comment (acu192) [reply packages/vue-app/template/utils.js:601]: What about if you call redirect like this?

```javascript
<script>
export default {
  middleware ({ redirect }) {
    const query = { foo: 'bar' }
    return redirect('https://nuxtjs.org/', query)
  }
}
</script>
```

I mean, if you pass a second parameter to `redirect()`, like above, which is the `query` parameter, which is being ignored now. Please correct me if I'm wrong.

Comment (clarkdo) [reply packages/vue-app/template/utils.js:601]: Ah, correct, thanks

**Review by pi0 — APPROVED** [review]

**Pull request merged**